src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityMiner.java

```diff
@@ -151,19 +131,44 @@ protected ModularUI createUI(EntityPlayer entityPlayer) {
                 .setMaxWidthLimit(84));
-
+        builder.widget(new SlotWidget(chargerInventory, 0, getTier() == 3 ? 88 : 79, 72)
+                .setBackgroundTexture(GuiTextures.SLOT, GuiTextures.CHARGER_OVERLAY));
 
         return builder.build(getHolder(), entityPlayer);
     }
 
+    private void addDisplayText(List<ITextComponent> textList) {
+        textList.add(new TextComponentString(String.format("sX: %d", this.minerLogic.getX().get())));
+        textList.add(new TextComponentString(String.format("sY: %d", this.minerLogic.getY().get())));
+        textList.add(new TextComponentString(String.format("sZ: %d", this.minerLogic.getZ().get())));
+        textList.add(new TextComponentString(String.format("Radius: %d", this.minerLogic.getCurrentRadius())));
+        if (this.minerLogic.isDone())
+            textList.add(new TextComponentTranslation("gregtech.multiblock.large_miner.done").setStyle(new Style().setColor(TextFormatting.GREEN)));
+        else if (this.minerLogic.isActive())
+            textList.add(new TextComponentTranslation("gregtech.multiblock.large_miner.working").setStyle(new Style().setColor(TextFormatting.GOLD)));
+        else
+            textList.add(new TextComponentTranslation("gregtech.multiblock.work_paused"));
+        if (isInventoryFull)
+            textList.add(new TextComponentTranslation("gregtech.multiblock.large_miner.invfull").setStyle(new Style().setColor(TextFormatting.RED)));
+        if (!drainEnergy(true))
+            textList.add(new TextComponentTranslation("gregtech.multiblock.large_miner.needspower").setStyle(new Style().setColor(TextFormatting.RED)));
+    }
+
+    private void addDisplayText2(List<ITextComponent> textList) {
+        textList.add(new TextComponentString(String.format("mX: %d", this.minerLogic.getMineX().get())));
+        textList.add(new TextComponentString(String.format("mY: %d", this.minerLogic.getMineY().get())));
+        textList.add(new TextComponentString(String.format("mZ: %d", this.minerLogic.getMineZ().get())));
+    }
+
     @Override
     public void addInformation(ItemStack stack, @Nullable World player, List<String> tooltip, boolean advanced) {
         tooltip.add(I18n.format("gregtech.universal.tooltip.voltage_in", energyContainer.getInputVoltage(), GTValues.VN[getTier()]));
         tooltip.add(I18n.format("gregtech.universal.tooltip.energy_storage_capacity", energyContainer.getEnergyCapacity()));
         tooltip.add(I18n.format("gregtech.machine.miner.tooltip"));
-        tooltip.add(I18n.format("gregtech.machine.miner.usage", getWorkingArea(), getWorkingArea(), getTick() / 20, getEnergyPerTick()));
+        tooltip.add(I18n.format("gregtech.machine.miner.usage", getWorkingArea(this.minerLogic.getMaximumRadius()), getWorkingArea(this.minerLogic.getMaximumRadius()), this.minerLogic.getSpeed() / 20, this.energyPerTick));
     }
 
+    @Override
     public boolean drainEnergy(boolean simulate) {
```

Comment:
For the energy container, you can call `energyContainer.drainEnergy(energyPerTick)`, It does the exact same thing, just looks easier to read.